### PR TITLE
Initial draft

### PR DESCRIPTION
**Pull request by mnibecker**
PR from issue #1

### CODE REVIEW

**Comment by mnibecker** [comment]
@go-oleg

File: src/inputs/actions.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+export const INPUT_DEFS_UPDATE = 'INPUT_DEFS_UPDATE'
+export const INPUT_VALUE_UPDATE = 'INPUT_VALUE_UPDATE'
+
+export function updateInput(input_id, value) {
+    return {
+        type: INPUT_VALUE_UPDATE,
+        input_id: input_id,
+        value: value
```
Comment (davidvgalbraith) [inline src/inputs/actions.js]: Can you just make this
```
input_id,
value
```
?

**Comment by davidvgalbraith** [comment]
Does the future have semicolons? It seems like half your lines have them and half of them don't.

File: src/inputs/input-group.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import _ from 'underscore';
+import React, { Component, PropTypes } from 'react';
+import { connect } from 'react-redux';
+import InputContainer from './components/input-container';
+
+class InputGroup extends Component {
+    _onKeyUp = (e) => {
+        if (e.keyCode === 13 && e.shiftKey) {
```
Comment (davidvgalbraith) [inline src/inputs/input-group.js]: `const ENTER_KEY = 13` and `if (e.keyCode === ENTER_KEY)` might aid comprehension here.

File: src/utils/api.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+
+const API_PREFIX = "/api/v0";
+
+export default class OutriggerHttp {
+    constructor(outriggerUrl) {
+        if (!outriggerUrl.startsWith('http')) {
+            throw new Error('Url scheme must be \'http\' or \'https\'');
+        }
+
+        this.outriggerUrl = outriggerUrl;
```
Comment (davidvgalbraith) [inline src/utils/api.js]: How about `this.url = ${this.outriggerUrl}${API_PREFIX}/` and then you can just use `fetch(${this.url}prepare)` etc. in the methods below.

File: src/inputs/actions.js (hunk 1 above)
Comment (mnibecker) [inline src/inputs/actions.js]: good call, senor

File: src/inputs/input-group.js (hunk 2 above)
Comment (mnibecker) [inline src/inputs/input-group.js]: Nice idea!

File: src/utils/api.js (hunk 3 above)
Comment (mnibecker) [inline src/utils/api.js]: yep

**Comment by mnibecker** [comment]
I wouldn't be so presumptuous to say I know what the future holds, but I think we're staying with semicolons. In a future pr I'll add a linting pass that'll make sure we got them semi-colons.

**Comment by mnibecker** [comment]
Made your changes @davidvgalbraith, you're quite a code reader!

File: example/index.html [hunk 4]
```diff
@@ -0,0 +1,20 @@
+<!DOCTYPE hmtl>
```
Comment (go-oleg) [inline example/index.html]: s/hmtl/html/

File: src/view/view-layout.js [hunk 5]
```diff
@@ -0,0 +1,75 @@
+import _ from 'underscore';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+
+import Sink from './sink';
+import sinkLayoutGen from './sink-layout-gen';
+
+class View extends Component {
```
Comment (go-oleg) [inline src/view/view-layout.js]: Not quite grasping the role of this component. Does it need to be its own thing or can it be folded into ViewLayout?

Comment (mnibecker) [inline src/view/view-layout.js]: The whole point of it is, that I basically need `ViewLayout` to get completely thrown away whenever the job_id changes. The `componentCharts` array needs to get reset and componentCharts need to get properly disposed. Doing it other way would require some propsDidChange bullshit.

File: example/index.html (hunk 4 above)
Comment (mnibecker) [inline example/index.html]: right right

File: src/view/view-layout.js (hunk 5 above)
Comment (go-oleg) [inline src/view/view-layout.js]: I see, can you pass the `job_id` as a `key` prop to ensure `ViewLayout` gets unmounted every time the `job_id` changes?

Comment (mnibecker) [inline src/view/view-layout.js]: doh, yeah I should probably do that right?

Comment (mnibecker) [inline src/view/view-layout.js]: fixed, good call @go-oleg

**Pull request merged**